packages/dev/loaders/src/glTF/2.0/Extensions/KHR_animation_pointer.ts

```diff
@@ -0,0 +1,343 @@
+            this._loader.babylonScene._blockEntityCollection = !!this._loader._assetContainer;
+            const group = new AnimationGroup(animation.name || `animation${animation.index}`, this._loader.babylonScene);
+            group._parentContainer = this._loader._assetContainer;
+            this._loader.babylonScene._blockEntityCollection = false;
+            animation._babylonAnimationGroup = group;
+        }
+        const babylonAnimationGroup = animation._babylonAnimationGroup;
+
+        const promises = new Array<Promise<any>>();
+        ArrayItem.Assign(animation.channels);
+        ArrayItem.Assign(animation.samplers);
+
+        for (const channel of animation.channels) {
+            promises.push(this._loadAnimationChannelAsync(`${context}/channels/${channel.index}`, context, animation, channel));
+        }
+
+        return Promise.all(promises).then(() => {
+            babylonAnimationGroup.normalize(0);
+            return babylonAnimationGroup;
+        });
+    }
+
+    /**
+     * @hidden Loads a glTF animation channel.
+     * @param context The context when loading the asset
+     * @param animationContext The context of the animation when loading the asset
+     * @param animation The glTF animation property
+     * @param channel The glTF animation channel property
+     * @param animationTargetOverride The babylon animation channel target override property. My be null.
+     * @returns A void promise when the channel load is complete
+     */
+    public _loadAnimationChannelAsync(
+        context: string,
+        animationContext: string,
+        animation: IAnimation,
+        channel: IAnimationChannel,
+        animationTargetOverride: Nullable<IAnimatable> = null
+    ): Promise<void> {
+        if (channel.target.path != AnimationChannelTargetPath.POINTER) {
+            throw new Error(`${context}/target/path: Invalid value (${channel.target.path})`);
+        }
+
+        if (channel.target.node != undefined) {
+            // According to KHR_animation_pointer specification
+            // If this extension is used, the animation.channel.target.node must not be set.
+            // Because the node isn’t defined, the channel is ignored and not animated due to the specification.
+            return Promise.resolve();
+        }
+
+        const pointer = channel.target.extensions?.KHR_animation_pointer?.pointer;
+        if (!pointer) {
+            throw new Error(`${context}/target/extensions.KHR_animation_pointer.pointer MUST be set.`);
+        }
+
+        const sampler = ArrayItem.Get(`${context}/sampler`, animation.samplers, channel.sampler);
+
+        return this._loadAnimationSamplerAsync(`${context}/samplers/${channel.sampler}`, sampler).then((data) => {
+            // this is where we process the pointer.
+            const animationTarget = this._parseAnimationPointer(`${context}/extensions/${this.name}/pointer`, pointer);
+
+            if (animationTarget) {

```

Comment:
I would recommend early out here to avoid indenting as much.

```suggestion
            if (!animationTarget) {
                return;
            }
```